src/quantuminspire/qiskit/backend_qx.py

```diff
@@ -248,41 +257,51 @@ def __validate_number_of_clbits(self, experiment: QasmQobjExperiment) -> None:
                                                  " number of qubits when using conditional gate operations")
 
     @staticmethod
-    def __validate_no_usage_after_measure(experiment: QasmQobjExperiment) -> None:
-        """ When using FSP (full state projection) certain instructions cannot be handled correctly because
-            intermediate measurements are not taking place in FSP. The measurements are postponed until the end of the
-            program.
-            Therefore some constructions are currently not supported:
-            1. When qubits are used again after being measured.
-            2. When conditional gate operations (binary controlled gates) are done based on previously measured
-               classical bits
+    def __validate_full_state_projection(experiment: QasmQobjExperiment) -> bool:
+        """ FSP (Full State Projection) can be used when no measurements are found in the circuit or when no
+            other gates are found after measurements.
 
         Args:
             experiment: The experiment with gate operations and header.
 
-        Raises:
-            QisKitBackendError: When a construction is used that is not supported.
+        Returns:
+            True when FSP can be used, otherwise False.
         """
-        measured_qubits = []
-        memory_bits = []
+        fsp = True
+        measurement_found = False
         for instruction in experiment.instructions:
             if instruction.name == 'measure':
-                for qubit in instruction.qubits:
-                    measured_qubits.append(qubit)
-                for qubit in instruction.memory:
-                    memory_bits.append(qubit)
+                measurement_found = True
             else:
```

Comment:
why not `elif measurement_found:` instead of `else: if measurement_found: `?

Reply:
ok